web/auth/auth.go

```diff
@@ -315,10 +314,16 @@ func login(c echo.Context) error {
 	if ok { // The user was already logged-in
 		sessionID = sess.ID()
 	} else if lifecycle.CheckPassphrase(inst, passphrase) == nil {
+		ua := user_agent.New(c.Request().UserAgent())
```

Comment:
The user-agent parsing is only needed in some case, and I would prefer to not make the other cases slower by always doing that.

Reply:
When a request is received by the stack, the first thing we do is to determine if the domain is the instance domain (like `example.mycozy.cloud`) or a domain for an app (like `example-drive.mycozy.cloud`). And, to do that, the stack loads the instance from CouchDB. The instance is often needed later, and to avoid calling again CouchDB, we are using a cache, and `middleware.GetInstance()` is the way to get the instance from this cache.

In our terminology, a middleware is a function called before the main handler of the request (`func login(c echo.Context) error` here). `middleware.GetInstance()` is not a middleware, just an helper function tied to the middleware `middleware.NeedInstance()`.